wait_time_estimator_optimized.rb

```diff
@@ -0,0 +1,26 @@
+# app/services/wait_time_estimator_optimized.rb
+#Estimated wait time (optimized) 
+#FR-03 #34
+
+class WaitTimeEstimator
+    CACHE_TTL = 30.seconds
+  
+    def self.estimate(location_id)
+      Rails.cache.fetch("wait_time_#{location_id}", expires_in: CACHE_TTL) do

```

Comment:
```suggestion
      Rails.cache.fetch("wait_time_#{location_id}", expires_in: CACHE_TTL) do
      heavy_estimation_logic(location_id)
```
this code looks functional, just need to accomplish the test cases.